### PR TITLE
Minor: Encapsulate type check in GroupValuesColumn, avoid panic

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/datafusion/pull/12269 from @jayzhan211

## Rationale for this change
I had some minor suggestions during review: https://github.com/apache/datafusion/pull/12269#pullrequestreview-2326132576

## What changes are included in this PR?
1. Move type checking into the same module that it needs to be kepy in sync with
2. return internal error than panic on bug

## Are these changes tested?
CI

### CODE REVIEW

File: datafusion/physical-plan/src/aggregates/group_values/column_wise.rs [hunk 1]
```diff
@@ -154,7 +187,9 @@ impl GroupValues for GroupValuesColumn {
                         let b = ByteGroupValueBuilder::<i64>::new(OutputType::Binary);
                         v.push(Box::new(b) as _)
                     }
-                    dt => todo!("{dt} not impl"),
+                    dt => {
+                        return internal_err!("{dt} not supported in GroupValuesColumn")
```
Comment (goldmedal) [inline datafusion/physical-plan/src/aggregates/group_values/column_wise.rs:191]:
```suggestion
                        return not_impl_err!("{dt} not supported in GroupValuesColumn")
```
It was todo!() previously, which I assume means we might implement it in the future. How about using `not_impl_err` instead?

**Review by goldmedal — APPROVED** [review]
Thanks @alamb it looks good to me. Only one minor suggestion for error handling.

File: datafusion/physical-plan/src/aggregates/group_values/column_wise.rs [hunk 2]
```diff
@@ -78,6 +79,38 @@ impl GroupValuesColumn {
             random_state: Default::default(),
         })
     }
+
+    /// Returns true if [`GroupValuesColumn`] supports for the specified schema
```
Comment (comphead) [inline datafusion/physical-plan/src/aggregates/group_values/column_wise.rs:83]:
```suggestion
    /// Returns true if [`GroupValuesColumn`] supported for the specified schema
```

File: datafusion/physical-plan/src/aggregates/group_values/column_wise.rs [hunk 3]
```diff
@@ -78,6 +79,38 @@ impl GroupValuesColumn {
...
+            .iter()
+            .map(|f| f.data_type())
+            .all(Self::supported_type)
+    }
+
+    /// Returns true if the specified data type is supported by [`GroupValuesColumn`]
+    fn supported_type(data_type: &DataType) -> bool {
+        matches!(
+            *data_type,
+            DataType::Int8
```
Comment (comphead) [inline datafusion/physical-plan/src/aggregates/group_values/column_wise.rs:96]: should we use `DataType::is_signed_integer` `DataType::is_unsigned_integer` 

we can also use `is_numeric` but it includes Decimals

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

File: datafusion/physical-plan/src/aggregates/group_values/column_wise.rs (hunk 3 above)
Comment (alamb) [reply datafusion/physical-plan/src/aggregates/group_values/column_wise.rs:96]: This list needs to be kept in sync with what special implementations are available in `GroupValuesColumns` -- I will add a comment to make that clearer

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by alamb** [comment]
Thank you for the review @jayzhan211

**Comment by alamb** [comment]
Thanks @comphead and @jayzhan211

**Pull request merged**